src/lifecycle.mjs

```diff
@@ -99,8 +99,14 @@ export function extract(strategy, worker, messageRouter, args = []) {
   return new Promise(async (resolve, reject) => {
     let numberOfMessages = 0;
     const type = "extraction";
+    const interval = setInterval(() => {
+      log(
+        `${strategy.module.name} extractor is running with ${numberOfMessages} messages pending`
+      );
+    }, 120_000);
```

Comment:
It has happened 2 out of 2 times that I was coding a feature and the github actions wouldn't close. They appeared to be hang. It was due to this promise not being resolved. It is resolved when all the messages are consumed but if some messages are lost the promise never resolves.

This log here can tell us if a promise is still pending. As long as the promise is pending this log will run.

Reply:
I bet these two issues are related: https://github.com/neume-network/strategies/issues/195#issuecomment-1207336249